app/models/concerns/illiadable.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+# Mixin for requests that should be sent to ILLiad for initial handling
+module Illiadable
+  def submit!
```

Comment:
Using modules that override methods on the class they are mixed into, isn't the easiet to read.  Is there a reason this is needed?  I see that HoldRecall#submit! already calls SubmitIlliadRequestJob can we modify Scan#submit to call SubmitIlliadRequestJob directly?

even better might be to convert this module to a new class (e.g. `IlliadRequest`) and have a method on the "Illiadable" requests like `#illiad_request`.  I think there is value in separating the concept of "The user made a request" and "We made this ILL request to fulfull the users request"